lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -466,7 +466,6 @@ export class UserAgentApplication {
     private acquireTokenHelper(account: Account, interactionType: InteractionType, isLoginCall: boolean, request?: AuthenticationParameters, resolve?: any, reject?: any): void {
         // Track the acquireToken progress
         this.cacheStorage.setItem(TemporaryCacheKeys.INTERACTION_STATUS, Constants.inProgress);
-        const scope = request.scopes ? request.scopes.join(" ").toLowerCase() : this.clientId.toLowerCase();
```

Comment:
Setting `clientId` as the only scope of a request is how you tell STS that you are requesting an ID token, as opposed to an access token. So the logic here is that if you don't provide `request.scopes`, request an ID token by setting `clientId` as the only scope.

Reply:
(there should probably be a comment above this line explaining that)

Reply:
Ah yes I forgot about the ID-token only flow, I had presumed that was just a normal request with scope of "openid" but I guess that returns an access token as well. But above change actually still preserves the logic you've implicated; the lack of any scopes in request.scopes is actually initialized to [this.clientId] within the ctor of the request object itself, so all downstream code can just directly interact with the request object and reference request.scopes, without having to track that in a separate variable. In this change, that scope variable was unused, since it was only pre-computed pass to the registerCallback function, which I refactored to accept "requestUid" instead. I think in general the request object could have a function like GetScopeString() that returns 'this.scopes.join(" ").toLowerCase()' for when that format is necessary.